src/picongpu/include/particles/ionization/byField/BSI/AlgorithmBSIStarkShifted.hpp

```diff
@@ -42,38 +41,44 @@ namespace particles
 namespace ionization
 {
 
-    /** \struct AlgorithmBSI
+    /** \struct AlgorithmBSIStarkShifted
      *
      * \brief calculation for the Barrier Suppression Ionization model
      */
-    struct AlgorithmBSI
+    struct AlgorithmBSIStarkShifted
     {
 
         /** Functor implementation
          *
          * \tparam EType type of electric field
-         * \tparam BType type of magnetic field
          * \tparam ParticleType type of particle to be ionized
          *
-         * \param bField magnetic field value at t=0
          * \param eField electric field value at t=0
          * \param parentIon particle instance to be ionized with position at t=0 and momentum at t=-1/2
          */
         template<typename EType, typename BType, typename ParticleType >
         HDINLINE void
-        operator()( const BType bField, const EType eField, ParticleType& parentIon )
+        operator()( const EType eField, ParticleType& parentIon )
         {
 
             const float_X protonNumber = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
             float_X chargeState = attribute::getChargeState(parentIon);
 
-            uint32_t cs = math::float2int_rd(chargeState);
-
-            /* ionization condition */
-            if (math::abs(eField) / ATOMIC_UNIT_EFIELD >= AU::IONIZATION_EFIELD_HYDROGEN[cs] && chargeState < protonNumber)
+            /* verify that ion is not completely ionized */
+            if (chargeState < protonNumber)
             {
-                /* set new particle charge state */
-                parentIon[boundElectrons_] -= float_X(1.0);
+                uint32_t cs = math::float2int_rd(chargeState);
+                /* ionization potential in atomic units */
+                const float_X iEnergy = GetIonizationEnergies<ParticleType>::type()[cs];
+                /* critical field strength in atomic units */
+                float_X critField = (math::sqrt(float_X(2.))-float_X(1.)) * math::pow(iEnergy,float_X(3./2.));
+
+                /* ionization condition */
+                if (math::abs(eField) / ATOMIC_UNIT_EFIELD >= critField)
+                {
+                    /* set new particle charge state */
+                    parentIon[boundElectrons_] -= float_X(1.0);
```

Comment:
offline discussed: This algorithm should only calculate how many electrons are created but should not manipulate the particle in the same moment.